bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -0,0 +1,444 @@
+        properties.put("productModel", sensiboSky.getProductModel());
+        properties.put("macAddress", sensiboSky.getMacAddress());
+
+        updateThing(editThing().withChannels(newChannels).withProperties(properties).build());
+    }
+
+    public List<Channel> createDynamicChannels(final SensiboSky sensiboSky) {
+        final List<Channel> newChannels = new ArrayList<>();
+        final ModeCapability capabilities = sensiboSky.getCurrentModeCapabilities();
+        final ChannelTypeUID modeChannelType = addChannelType(SensiboBindingConstants.CHANNEL_TYPE_MODE, "Mode",
+                "String", sensiboSky.getRemoteCapabilities().keySet(), null, null);
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_MODE), "String")
+                .withType(modeChannelType).build());
+
+        final ChannelTypeUID swingModeChannelType = addChannelType(SensiboBindingConstants.CHANNEL_TYPE_SWING_MODE,
+                "Swing Mode", "String", capabilities.swingModes, null, null);
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_SWING_MODE), "String")
+                .withType(swingModeChannelType).build());
+
+        final ChannelTypeUID fanLevelChannelType = addChannelType(SensiboBindingConstants.CHANNEL_TYPE_FAN_LEVEL,
+                "Fan Level", "String", capabilities.fanLevels, null, null);
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_FAN_LEVEL), "String")
+                .withType(fanLevelChannelType).build());
+
+        final ChannelTypeUID targetTemperatureChannelType = addChannelType(
+                SensiboBindingConstants.CHANNEL_TYPE_TARGET_TEMPERATURE, "Target Temperature", "Number:Temperature",
+                sensiboSky.getTargetTemperatures(), "%d %unit%", "TargetTemperature");
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_TARGET_TEMPERATURE),
+                        "Number:Temperature")
+                .withType(targetTemperatureChannelType).build());
+
+        return newChannels;
+    }
+
+    private ChannelTypeUID addChannelType(final String channelTypePrefix, final String label, final String itemType,
+            final Collection<?> options, @Nullable final String pattern, @Nullable final String tag) {
+        final ChannelTypeUID channelTypeUID = new ChannelTypeUID(SensiboBindingConstants.BINDING_ID,
+                channelTypePrefix + getThing().getUID().getId());
+        final List<StateOption> stateOptions = options.stream()
+                .map(e -> new StateOption(e.toString(), e instanceof String ? beautify((String) e) : e.toString()))
+                .collect(Collectors.toList());
+        @SuppressWarnings("deprecation")
```

Comment:
What's the problem with using the "new" method?

Reply:
Done